### ISSUE TITLE
Support custom commands with code arguments

### ISSUE THREAD

**Issue by vries** [nmoroze/tclint#121]
I'm in the process of making the gdb testsuite tclint-clean.

I'm decided to do this one directory at at time, started with the one that has the fewest warnings, and so on.

Eventually I got to directories with many warnings, and manually fixing those became problematic.  So I wrote a simple sed script to automatically fix tclint problems.

After using it, I realized that it fixed problems in parts of the code tclint doesn't complain about.

This is due to the fact that we have many custom constructs that take in code that tclint doesn't know about.

For instance, we have proc_with_prefix.  That doesn't translate well to  a standalone reproducer, so I modified it to the following:
```
$ cat test.tcl
#!/usr/bin/tclsh

proc log { name body } {
    puts $name
    uplevel 1 $body
}

proc proc_with_log {name arguments body} {
    proc $name $arguments [list log $name $body]
}

set a 1
proc foo {} {
    puts [expr 1 + $::a]
}

proc_with_log foo_with_log {} {
    puts [expr 2 + $::a]
}

foo
foo_with_log
$ ./test.tcl
2
foo_with_log
3
$  tclint test.tcl
test.tcl:9:5: expected braced word or word without substitutions in argument interpreted as list [command-args]
test.tcl:14:16: expression with substitutions should be enclosed by braces [unbraced-expr]
```
The unbraced-expr is only triggered for proc foo, not for proc foo_with_log.

I wonder if it might be too heavy handed to construct a gdb testsuite dialect plugin.

Perhaps we could use the configuration file to indicate what commands are aliases of other commands in terms of parsing and checking.

But we also have things like:
```
proc with_test_prefix { prefix body } {
    ...
}

with_test_prefix foo {
    puts [expr 2 + $a]
}
```
and since unlike proc_with_prefix this wasn't designed to mimic an existing tcl construct,  I don't know if there's an alias available for that.

**Comment by vries** [comment]
> I wonder if it might be too heavy handed to construct a gdb testsuite dialect plugin.

Hmm, I suppose it could be trivial to define an alias in python, using a dynamic plugin.  No documentation or example for that though, AFAICT, so I'm not sure where to start.

**Comment by vries** [comment]
I realized that by ignoring the plugin concept and just modifying the tclint source code, I could get to a result faster.

So I wrote [this patch](https://github.com/user-attachments/files/22460095/tmp.patch),  which handles things like proc_with_prefix and with_test_prefix (for now, only caring about number of arguments and script arguments).

I had to add some logic to prevent for instance the definition of proc_with_prefix triggering a redefined-builtin violation.

**Comment by nmoroze** [comment]
Done! https://github.com/nmoroze/tclint/releases/v0.6.2

**Comment by vries** [comment]
> Done! https://github.com/nmoroze/tclint/releases/v0.6.2

Thanks :)
